apps/monitoring/splunk/custom/api.pm

```diff
@@ -0,0 +1,395 @@
+        $self->{output}->option_exit();
+    }
+
+    sleep(1.5);
+
+    my $query_status = $self->request_api(
+        method => 'GET',
+        endpoint => '/services/search/jobs/' . $query_sid->{sid},
+    );
+
+    foreach (@{$query_status->{content}->{'s:dict'}->{'s:key'}}) {
+        if ($_->{name} eq 'isDone' && $_->{content} == 0){
+            $self->{output}->add_option_msg(short_msg => "Search command wasn't completed.");
+            $self->{output}->option_exit();
+        } elsif ($_->{name} eq 'isFailed' && $_->{content} == 1) {
+            $self->{output}->add_option_msg(short_msg => "Search command failed.");
+            $self->{output}->option_exit();  
+        }
+    }
+
+    my $query_res = $self->request_api(
+        method => 'GET',
+        endpoint => '/services/search/jobs/' . $query_sid->{sid} . '/results',
+    );
+    my $query_count = $query_res->{result}->{field}->{value}->{text};
+
+    return $query_count;
+}
+
+sub request_api {
+    my ($self, %options) = @_;
+
+    if (!defined($self->{session_token})) {
+        $self->{session_token} = $self->get_access_token(statefile => $self->{cache});
+    }
+    $self->settings();
+    
+    my $content = $self->{http}->request(
+        method => $options{method},
+        url_path => $options{endpoint},
+        post_param => $options{post_param},
+        unknown_status => $self->{unknown_http_status},
+        warning_status => $self->{warning_http_status},
+        critical_status => $self->{critical_http_status}
+    );
+
+    if ($self->{http}->get_code() < 200 || $self->{http}->get_code() >= 300) {
+        $self->clean_session_token(statefile => $self->{cache});     
+        $self->get_access_token(statefile => $self->{cache});
+        $self->settings();
+        $content = $self->{http}->request(
+            method => $options{method},
+            url_path => $options{endpoint},
+            post_param => $options{post_param},
+            unknown_status => $self->{unknown_http_status},
+            warning_status => $self->{warning_http_status},
+            critical_status => $self->{critical_http_status}
+        );
+    }
+
+    my $xml_result;
+    eval {
+        $SIG{__WARN__} = sub {};
+        $xml_result = XMLin($content, ForceArray => $options{force_array}, KeyAttr => []);
+    };
+    if ($@) {
+        $self->{output}->add_option_msg(short_msg => "Cannot decode xml response: $@");
+        $self->{output}->option_exit();
+    }
+    if (defined($xml_result->{XPathError})) {
+        $self->{output}->add_option_msg(short_msg => "Api return error: " . $xml_result->{XPathError}->{Reason});
+        $self->{output}->option_exit();
+    }
+
+    return $xml_result;
+}
+
+1;
+
+__END__
+
+=head1 NAME
+
+Splunk API.
+
+=head1 SYNOPSIS
+
+Splunk API custom mode.
+
+=head1 REST API OPTIONS
```

Comment:
Please rename `REST API OPTIONS` to `XMLAPI OPTIONS`. Otherwise the help is not displayed.